main/autogen/autogen.cc

```diff
@@ -262,6 +264,10 @@ class AutogenWalk {
             // RHS. Mark this `Definition` as an alias for it.
             ENFORCE(refMap.count(original.rhs.get()));
             def.aliased_ref = refMap[original.rhs.get()];
+        } else if (autogenVersion >= 3 && lhs->symbol.exists() && lhs->symbol.data(ctx)->isTypeAlias()) {
+            // if the LHS has already been annotated as a type alias by the namer, the definition is (by definition,
+            // hah) a type alias.
+            def.type = Definition::Type::TypeAlias;
```

Comment:
Main change is here. Added a version check as well, as we only expose this in v3.

Reply:
I think it'd be a much cleaner change if you keep the knowledge of version out of the generator and have it only operate in the world of the current max version. Would it be possible to instead have the msg pack rendering output a casgn instead of typealias if version < 3?

Reply:
Hmm. Can look into this

Reply:
Fixed